python/pyspark/cloudpickle.py

```diff
@@ -220,12 +322,7 @@ def save_function(self, obj, name=None):
 
         if name is None:
             name = obj.__name__
-        try:
-            # whichmodule() could fail, see
-            # https://bitbucket.org/gutworth/six/issues/63/importing-six-breaks-pickling
-            modname = pickle.whichmodule(obj, name)
-        except Exception:
-            modname = None
+        modname = pickle.whichmodule(obj, name)
```

Comment:
It's a good question, the underlying issue was marked resolved in 2014 and from looking at the commit it seems like it should actually be resolved. That being said its true some people might be on a system installed with an old verison of six so perhaps we should keep this workaround.